docs/github-provider.md

```diff
@@ -0,0 +1,100 @@
+# How to use the *GitHub Provider*
+
+## Setup the provider
+
+To setup the provider you need to do 2 main things:
+- get an access token from your GitHub account
+- set the access token in your development environment
+
+### Access token
+
+- Go to https://github.com/settings/tokens/ and click on "Generate new token" button. (you can choose either a "Fine-grained token" or a "Classic token").
+- After generating the token, keep it preciously.
+
+### Set the token
+
+- Create a new file named ```.env``` at the root of the project and write this line: ```GITHUB_TOKEN="<token_generated>"```. 
+  Replace ```<token_generated>``` with the token you just copied from GitHub.
+- Create a new file with the name of your choice but with the extension ```.sh```.
+  For example: ```.env.sh```
+- Finally run this command ```source .env.sh```.
+  This command allow you to export the ```GITHUB_TOKEN``` environment variable and use it in the Sismo Hub application.
+
+
+You can now use the GitHub Provider!
+
+
+---
+
+## Use the provider
+
+### Methods
+
+There are 2 main methods in the GitHub Provider:
+- ```getRepositoriesContributors```
+- ```getRepositoriesStargazers```
+
+<br>
+
+#### getRepositoriesContributors
+
+**This method allow you to fetch all the contributors of one or more GitHub repositories.**
+
+There are 3 arguments to give to this method:
+- ```repositories```
+- ```getOrganizationMembers```
+- ```defaultValue```
+
+#### getRepositoriesStargazers
+
+**This method allow you to fetch all the GitHub user who put a star on one or more GitHub repositories.**
+
+There are 2 arguments to give to this method:
+- ```repositories```
+- ```defaultValue```
+
+<br>
+
+#### repositories
+
+```repositories``` is the array that contains all the repository you want to fetch.
+For example, if you want to fetch all the contributors of the Solidity GitHub repository from Ethereum, you need to pass this argument: ```["ethereum/solidity"]```
+
+#### defaultValue
+
+```defaultValue``` define the value of all the items of the method returned object.
+The object returned by the methods is of this form:
+```
+{
```

Comment:
The correct form is `"github:username:id": 2`. You can explain simply the username and the id